src/mpid/ch4/shm/ipc/gpu/gpu_post.c

```diff
@@ -151,13 +151,21 @@ int MPIDI_GPU_get_ipc_attr(const void *vaddr, int rank, MPIR_Comm * comm,
 
     recv_lrank = MPIDI_GPUI_global.local_ranks[MPIDIU_rank_to_lpid(rank, comm)];
     ipc_attr->ipc_type = MPIDI_IPCI_TYPE__GPU;
+    ipc_attr->threshold.send_lmt_sz = MPIR_CVAR_CH4_IPC_GPU_P2P_THRESHOLD;
 
     local_dev_id = MPL_gpu_get_dev_id_from_attr(&ipc_attr->gpu_attr);
     int global_dev_id = MPL_gpu_local_to_global_dev_id(local_dev_id);
 
     mpl_err = MPL_gpu_get_buffer_bounds(vaddr, &pbase, &len);
     MPIR_ERR_CHKANDJUMP(mpl_err != MPL_SUCCESS, mpi_errno, MPI_ERR_OTHER, "**gpu_get_buffer_info");
 
+    /* Don't create or search for an ipc handle if the buffer doesn't meet the threshold
+     * requirement */
+    if (ipc_attr->threshold.send_lmt_sz > len) {
+        ipc_attr->ipc_type = MPIDI_IPCI_TYPE__NONE;
+        goto fn_fail;
+    }
```

Comment:
The lazy mechanism (https://github.com/pmodels/mpich/pull/6417/commits/97db11e901f57126ce0c63fe155b67daab6ebc4d) will allow ipc path even when it is below the threshold.

Didn't we already check the threshold in `ipc_send.h`? The buffer bounds (`len`) here should be only larger than the actual data size, so I don't see why we need the test here.

Reply:
The reason this check was added here is because there were no safe guards for creating IPC handles if they didn't meet the threshold requirement. This is an issue for ZE because they are backed by file descriptors which are a limited resource.

Now that the lazy mechanism exists, I actually moved the lazy check to this spot (see 14ab299e4ae842d114d3950412469ca42817cf31) to preserve the safe guard mentioned above.

If it makes more sense, I can merge this commit and 14ab299e4ae842d114d3950412469ca42817cf31

Reply:
I squashed the two changes and reworded the commit (9131542c7d833e335a0d2769252c4bf6a59cdb3f)